src/test/kotlin/com/igorwojda/integer/power/challenge.kt

```diff
@@ -2,9 +2,25 @@ package com.igorwojda.integer.power
 
 import org.amshove.kluent.shouldBeEqualTo
 import org.junit.jupiter.api.Test
+import kotlin.reflect.KFunction
+
+//private fun power(base: Int, exponent: Int): Int {
+//    var result = base
+//    var pow = exponent
+//    if(exponent==0){
+//        return 1
+//    }
+//    while(pow>1){
+//        result*=base
+//        pow--
+//    }
+//    return result
+//}
 
 private fun power(base: Int, exponent: Int): Int {
-    TODO("not implemented")
+    operator fun List<Int>.invoke() = reduce { it, acc -> it * acc }
```

Comment:
This is much closer to what i had in mind. To produce cleaner code you could do: `return List(exponent) { base }.reduce(Int::times)`